src/targets/registry.ts

```diff
@@ -408,20 +427,67 @@ export class RegistryTarget extends BaseTarget {
         this.logger.warn(
           `No files found that match "${onlyIfPresentPattern.toString()}", skipping the target.`
         );
+        this.published = true;
         return undefined;
       }
     }
 
-    const remote = this.registryConfig.registryRemote;
-    const username = await getAuthUsername(this.github);
-    remote.setAuth(username, getGithubApiToken());
+    if (RegistryTarget.lock) {
+      await RegistryTarget.lock;
+    }
 
-    await withTempDir(
-      directory =>
-        this.pushVersionToRegistry(directory, remote, version, revision),
-      true,
-      'craft-release-registry-'
-    );
-    this.logger.info('Release registry updated');
+    await (RegistryTarget.lock = this.doPublish(version, revision));
+    RegistryTarget.lock = undefined;
+  }
+
+  private async doPublish(version: string, revision: string) {
+    if (!RegistryTarget.localRepo) {
+      await withTempDir(
+        async dir => {
+          RegistryTarget.localRepo = {
+            dir,
+            git: await this.cloneRegistry(dir),
+          };
+        },
+        // We will clean the directory after pushing
+        false,
+        'craft-release-registry-'
+      );
+    }
+    let localRepo: LocalRegistry;
+    if (RegistryTarget.localRepo) {
+      localRepo = RegistryTarget.localRepo;
+    } else {
+      // XXX(BYK): This should NEVER happen
+      throw new Error(
+        `Local registry missing, it should have been cloned at this stage!`
+      );
+    }
+    this.commitVersionToRegistry(localRepo, version, revision);
+
+    if (this.isLastToPublish()) {
```

Comment:
Please also add documentation around execution order. It looks like the _last_ target triggers the actual publish. While it makes sense, it might not necessarily be what a user expects. (knowing how it's implemented it's the only sensible approach though)